packages/core/lib/commands/exec.js

```diff
@@ -73,29 +73,16 @@ const command = {
 
         // `--compile`
         if (options.c || options.compile) {
-          return Contracts.compile(config, function(err) {
-            if (err) return done(err);
-
-            Require.exec(
-              config.with({
-                file: file
-              }),
-              done
-            );
-          });
+          return Contracts.compile(config);
```

Comment:
But `Contracts.compile` isn't async, is it?

Reply:
It sure is! It actually has both apis right now (callback or Promise).

Reply:
Want to try updating to the new compile interface? I ask because I'd like someone else besides me to get familiarity with it, but do feel free to tell me to do this work myself

Reply:
Sure thing! I'll chat with you later to talk about the requirements.